tile_map/src/tile_map_view.cpp

```diff
@@ -214,7 +214,7 @@ namespace tile_map
 
     for (size_t i = 0; i < precache_.size(); i++)
     {
-      TexturePtr texture = precache_[i].texture;
+      TexturePtr& texture = precache_[i].texture;
```

Comment:
It isn't clear to me how this would change the behavior of the code here.

The original code is copying the shared ptr, but that is relatively cheap and shouldn't invoke a genTexture call unless the shared ptr is null.  The new code avoids the ptr copy but shouldn't reduce genTexture calls.

Is there something else going on here?

Reply:
Ah, I see the issue.  Without the reference the new texture object created isn't saved back to the cached objects.